### PR TITLE
initial hosts file parsing support

### PR DESCRIPTION
**Pull request by banaru4**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -8,6 +8,9 @@ cfg-if = "0.1"
 serde = "1.0"
 serde_derive = "1.0"
 serde_json = "1.0"
+regex = "1.0.0"
+lazy_static = "1.0.0"
+prettytable-rs = "^0.6"
```
Comment (awakecoding) [inline Cargo.toml]: please remove if unused in this PR

File: src/common/hosts.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        lazy_static! {
+        //regex filter: remove everything from "#" till the line break
+        static ref RE: Regex = Regex::new(r"(?m)^([^#]*)").unwrap();
+    }
+
+        for line in system_reader.get_hosts_file().unwrap().lines() {
+            let captures = RE.captures(&line);
+            if let Some(cap) = captures {
+                //omitting empty outputs from regex
+                if cap.get(0).unwrap().as_str().len() == 0 {
```
Comment (awakecoding) [inline src/common/hosts.rs]: unwrap() should be handled gracefully

File: src/common/hosts.rs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                    continue
+                }
+                    else {
+                        let mut etc_hosts = EtcHosts::new();
+                        let v: Vec<_> = cap.get(0).unwrap().as_str().trim().split_whitespace().collect();
```
Comment (awakecoding) [inline src/common/hosts.rs]: unwrap() should be handled gracefully

File: src/lib.rs [hunk 4]
```diff
@@ -5,9 +5,16 @@ extern crate serde;
 extern crate serde_derive;
 #[macro_use]
 extern crate serde_json;
+extern crate regex;
+#[macro_use]
+extern crate lazy_static;
+//#[macro_use]
+//extern crate prettytable;
```
Comment (awakecoding) [inline src/lib.rs]: remove commented out externs

File: src/windows/mod.rs [hunk 5]
```diff
@@ -56,23 +58,29 @@ impl SystemReaderInterface for SystemReader {
...
+
     fn get_wmi_nicconfig_details(&self) -> Option<String> {
         let output = Command::new("wmic")
             .args(&["nicconfig", "get", "IPEnabled,InterfaceIndex,MACAddress,MTU", "/format:list"]).output().ok()?;
         String::from_utf8(output.stdout).ok()
     }
+
+    fn get_hosts_file(&self) -> Option<String> {
+        let mut s = String::new();
+        File::open("c:\\windows\\system32\\drivers\\etc\\hosts").ok()?.read_to_string(&mut s).ok()?;
```
Comment (awakecoding) [inline src/windows/mod.rs]: C:\Windows\System32 should be dynamically detected from either an environment variable like %SYSTEMROOT%\System32

Comment (ekse) [reply src/windows/mod.rs]: std:env::var could be used to get the value of SYSTEMROOT.
https://doc.rust-lang.org/std/env/fn.var.html

File: src/common/hosts.rs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    pub fn get_hosts (system_reader: &SystemReaderInterface) -> Vec<EtcHosts> {
+
+        let mut hosts: Vec<EtcHosts> = Vec::new();
+
+        lazy_static! {
+        //regex filter: remove everything from "#" till the line break
+        static ref RE: Regex = Regex::new(r"(?m)^([^#]*)").unwrap();
+    }
```
Comment (ekse) [inline src/common/hosts.rs]: Move the regex outside of the function and give a descriptive name.

File: src/common/hosts.rs [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+                                etc_hosts.hostnames = v[1..].join(",");
+                                hosts.push(etc_hosts);
+                            }
+                            Err(_e) => {
+                                continue;
+                            }
+                        };
+                    }
+            }
+        }   hosts
```
Comment (ekse) [inline src/common/hosts.rs]: The indentation needs to fixed.

**Pull request merged**